### ISSUE TITLE
Removing candidate from sub list not having intended effect

### ISSUE THREAD

**Issue by samschlicht** [talentbeyondboundaries/tbbtalentv2#1632]
Reported by Naji, this looks like a bug.

It concerns a seeming reappearance of an issue that he reported to us nearly a year ago, which appeared to be fixed in #543.

To summarise:

- A candidate was mistakenly added to [the submission list of a 2022 version of a job](https://tctalent.org/admin-portal/list/4720), instead of [the 2024 version](https://tctalent.org/admin-portal/list/10340), as was intended.
- Melody (MAP) the admin user removed the opportunity but it did not have the expected effect, which is that it would stay on the list but as a closed opportunity with a closed stage.
- Instead we have [a SF candidate opp](https://talentbeyondboundaries.lightning.force.com/lightning/r/Opportunity/006Uu00000APuZ7IAL/view) and[ a TC case](https://tctalent.org/admin-portal/opp/11134), both at Prospect stage, but with no entry, closed or otherwise, on [the 2022 TC submission list](https://tctalent.org/admin-portal/list/4720).

[Here's the Slack convo between me and Naji](https://refugeejobsmarket.slack.com/archives/C7QK72T4J/p1731385375360989) — whose useful content thus far is summarised above.

**Comment by samschlicht** [comment]
Firstly this is not really related to #543, which fixed an issue where the Remove button on submission lists wasn't closing out SF candidate opps.

What appears to have happened here is that a candidate opportunity was created in SF and the TC, but the candidate was never added to the list — there is no entry in the candidate_saved_list table for this combination of list and candidate, which there would be even if it were closed.

**Comment by samschlicht** [comment]
I'm following the thread, and it is the case that candidate opps are created on the TC and SF _before_ the new candidate list entry is saved — but the code is working at the moment.

**Comment by samschlicht** [comment]
Try as I have, I just can't seem to replicate this bug! However, looking at the logs, I think I can see what has happened:

- As I've already noted, when adding candidates to a list, the TC and SF opps are created first. This happened at 2245 AST:


<img width="775" alt="Screenshot 2024-11-21 at 4 46 53 pm" src="https://github.com/user-attachments/assets/c049ebf3-c913-4bb3-b141-257579a54bde">


- At 2246, the DB connection is lost and one of our services reboots:

<img width="843" alt="Screenshot 2024-11-21 at 4 47 39 pm" src="https://github.com/user-attachments/assets/e555f253-3b94-4368-99e0-00a6864c57e2">


- For 40 mins before this, this log appears 100s of times:


<img width="878" alt="Screenshot 2024-11-21 at 4 54 15 pm" src="https://github.com/user-attachments/assets/3b022e2e-620a-4f7c-ae2f-64fe7479f000">


So, basically, there were a lot of errors preceding this action, and DB dropout during it. It succeeded in creating the opps but not adding the candidate to the list.

I'm going to look a bit further in the logs to see if I can find a root cause, but I think I can at least safely say that this is not a code issue and hopefully won't reappear. 

I can also close out the candidate opp, since I no longer need it in its previous state to investigate. To our newly added stage!

<img width="497" alt="Screenshot 2024-11-21 at 5 38 57 pm" src="https://github.com/user-attachments/assets/c38b6db6-50dc-4890-86e3-eabe8420c047">

**Comment by samschlicht** [comment]
Interestingly the heap blew that morning, and there seem to have been issues all day:

<img width="615" alt="Screenshot 2024-11-21 at 6 32 15 pm" src="https://github.com/user-attachments/assets/adc3edee-d612-4c8b-8387-16953c22126a">

It happens while the job chat notification emails and search watcher emails are being sent:

<img width="1168" alt="Screenshot 2024-11-21 at 6 34 57 pm" src="https://github.com/user-attachments/assets/e25ae56f-bdd6-43b9-8ce7-30632579f6c3">
 
Basically it's running quite a few searches and attempting to send emails but getting mostly errors, then the heap blows.

Might be worth looking at other occasions when these emails are being sent, and also whether the automatic task deployment took place.
